### PR TITLE
Pattern v2

### PR DESCRIPTION
**Pull request by bandvov**

### CODE REVIEW

File: src/pages/pattern/pattern-details/pattern-details.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    images: PropTypes.shape({
+      thumbnail: PropTypes.string
+    }),
+    material: PropTypes.string,
+    name: PropTypes.arrayOf(valueShape)
+  })
+};
+
+PatternDetails.defaultProps = {
+  match: {},
```
Comment (Alan18081) [inline src/pages/pattern/pattern-details/pattern-details.js:58]: I think `match` should be required

File: src/containers/table-container-row/table-container-row.js [hunk 2]
```diff
@@ -19,9 +20,9 @@ const TableContainerRow = ({
 
   const dense = useSelector(({ Table }) => Table.dense);
 
-  const propetries = { ...rest };
-  const tableCells = Object.values(propetries).map((propetry, index) => (
-    <TableCell key={index}>{propetry}</TableCell>
+  const properties = { ...rest };
+  const tableCells = Object.values(properties).map((property, index) => (
+    <TableCell key={index}>{`${property}`}</TableCell>
```
Comment (Alan18081) [inline src/containers/table-container-row/table-container-row.js:25]: There is no need to do string interpolation here, you can just use {property} without any problems

Comment (Alan18081) [reply src/containers/table-container-row/table-container-row.js:25]: Don't use `index` as a key, use `property` instead if it is always unique and stable or find some another value for the index, please

File: src/containers/table-container-head/table-container-head.js [hunk 3]
```diff
@@ -5,7 +5,9 @@ import { TableHead, TableRow, TableCell } from '@material-ui/core';
 
 const TableContainerHead = ({ titles }) => {
   const headRow = titles.map((title, index) => (
-    <TableCell key={index}>{title}</TableCell>
+    <TableCell data-cy={title} key={index}>
```
Comment (Alan18081) [inline src/containers/table-container-head/table-container-head.js:8]: Don't use `index` as a key, use `title` instead if it is always unique and stable or find some another value for the index, please

File: src/pages/pattern/pattern-add/pattern-add.js [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+      handmade: false
+    },
+    validationSchema: patternValidationSchema,
+    validateOnBlur: true,
+    onSubmit: (values) => {
+      const pattern = createPattern(values);
+      dispatch(addPattern({ pattern, upload }));
+    }
+  });
+  const TabPanels =
```
[the remark this answers is not in the document]
Comment (bandvov) [reply src/pages/pattern/pattern-add/pattern-add.js:91]: fixed

File: src/redux/pattern/pattern.sagas.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+} from '../snackbar/snackbar.actions';
+
+const {
+  SUCCESS_ADD_STATUS,
+  SUCCESS_DELETE_STATUS,
+  SUCCESS_UPDATE_STATUS
+} = config.statuses;
+
+export function* handlePatternsLoad({
+  payload = {
```
[the remark this answers is not in the document]
Comment (bandvov) [reply src/redux/pattern/pattern.sagas.js:43]: I used news as example. fixed

File: src/containers/table-container-row/table-container-row.js (hunk 2 above)
Comment (bandvov) [reply src/containers/table-container-row/table-container-row.js:25]: fixed

File: src/pages/pattern/pattern-details/pattern-details.js (hunk 1 above)
Comment (bandvov) [reply src/pages/pattern/pattern-details/pattern-details.js:58]: fixed

File: src/containers/table-container-head/table-container-head.js (hunk 3 above)
Comment (bandvov) [reply src/containers/table-container-head/table-container-head.js:8]: fixed

File: src/pages/pattern/pattern-add/pattern-add.js [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+        setPatternImage(e.target.result);
+      };
+      reader.readAsDataURL(e.target.files[0]);
+      setUpload(e.target.files[0]);
+    }
+  };
+
+  return (
+    <div className={styles.container}>
+      <form onSubmit={handleSubmit}>
```
[the remark this answers is not in the document]
Comment (bandvov) [reply src/pages/pattern/pattern-add/pattern-add.js:180]: fixed

File: src/pages/pattern/pattern-details/pattern-details.js (hunk 1 above)
Comment (bandvov) [reply src/pages/pattern/pattern-details/pattern-details.js:58]: fixed

File: src/pages/pattern/pattern-add/pattern-add.js [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+            type='submit'
+            title='Зберегти'
+          />
+        </div>
+        {languages.length > 0 ? (
+          <div>
+            <Grid item xs={12}>
+              <Paper className={styles.patternItemAdd}>
+                <label htmlFor='patternImage'>
+                  <Avatar
```
[the remark this answers is not in the document]
Comment (bandvov) [reply src/pages/pattern/pattern-add/pattern-add.js:197]: fixed

**Review by TorskyM — APPROVED** [review]

**Review by TorskyM — COMMENTED** [review]
Remove code smells

**Review by maksym-strus — APPROVED** [review]

**Pull request merged**